[Dependency Update] Update `a8c-ci-toolkit` to 3.4.2

ParaskP7 opened the pull request:
### Description

This PR updates `a8c-ci-toolkit` to [3.4.2](https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/releases/tag/3.4.2).

FYI: The breaking change in `3.0.0` is due to removing the `nvm_install` command, but this repo doesn't use that anywhere anyway. As such, it is safe to apply this update.

### Testing Instructions

- Make sure CI is green (🟢).

ParaskP7 committed:
Deps: Update a8c ci toolkit to 3.4.2

Release Notes: https://github.com/Automattic/
a8c-ci-toolkit-buildkite-plugin/releases/tag/3.4.2

------------------------------------------------------------------------

FYI: The breaking change in '3.0.0' is due to removing the 'nvm_install'
command, but this repo doesn't use that anywhere anyway. As such, it is
safe to apply this update.

For more info see: https://github.com/wordpress-mobile/
WordPress-Utils-Android/pull/148/files#r1666828195

AliSoftware commented:
Same as https://github.com/Automattic/pocket-casts-android/pull/2487#pullrequestreview-2174418571, would love to use the opportunity to start using `shared-pipeline-vars` in this repo 🙂

ParaskP7 committed:
CI: Use shared buildkite pipeline vars file

For more info see: https://github.com/wordpress-mobile/
WordPress-Utils-Android/pull/150#issuecomment-2225298878

ParaskP7 commented:
FYI @AliSoftware that this is now done: 74097ab
